app/src/main/java/agency/five/codebase/android/movieapp/ui/main/MainScreen.kt

```diff
@@ -0,0 +1,185 @@
+package agency.five.codebase.android.movieapp.ui.main
+
+import agency.five.codebase.android.movieapp.navigation.NavigationItem
+import agency.five.codebase.android.movieapp.ui.theme.LocalSpacing
+import agency.five.codebase.android.movieapp.R
+import agency.five.codebase.android.movieapp.navigation.MOVIE_ID_KEY
+import agency.five.codebase.android.movieapp.ui.favorites.FavoritesRoute
+import agency.five.codebase.android.movieapp.ui.home.HomeRoute
+import agency.five.codebase.android.movieapp.ui.moviedetails.MovieDetailsRoute
+import agency.five.codebase.android.movieapp.ui.theme.Blue
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.material.*
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.ColorFilter
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import androidx.navigation.NavDestination
+import androidx.navigation.NavType
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.currentBackStackEntryAsState
+import androidx.navigation.compose.rememberNavController
+import androidx.navigation.navArgument
+
+@Composable
+fun MainScreen() {
+    val navController = rememberNavController()
+    val navBackStackEntry by navController.currentBackStackEntryAsState()
+    var showBottomBar by remember {
+        mutableStateOf(true)
+    }
+    var showBackIcon by remember {
+        mutableStateOf(!showBottomBar)
+    }
```

Comment:
This solution doesn't scale well. Imagine that you need to add more screens, than you'll have to put the logic for showing bottomBar/backIcon into every screen: 

```kotlin
composable(someRoute) {
  showBottomBar = true/false
  showBackIcon = false/true
}
```

You can use provided `navBackStackEntry` for calculating showBottomBar:

```kotlin
val showBottomBar = when (navBackStackEntry?.destination?.route) {
    MovieDetailsDestination.route -> false
    else -> true
}

val showBackIcon = !showBottomBar
```